app/computing/domain_commands.py

```diff
@@ -46,7 +46,25 @@ def get_identifier(cls) -> str:
         return 'REGISTER'
 
     def invoke(self, receiver: base.SubproblemPool) -> List[Command]:
+        if self.identifier in receiver.results:
+            result = receiver.results[self.identifier]
+            result_command = BaseResultCommand(self.identifier, result)
+            return [result_command]
+
+        if self.identifier in receiver.in_progress_pool and self.identifier == receiver.current_subproblem_id:
+            if self._is_sender_priority_greater():
+                receiver.signal_subproblem_stop()
+                return []
+            else:
+                drop_command = BaseDropCommand(self.identifier)
+                return [drop_command]  
+
         return []
+    
+    def _is_sender_priority_greater(self):
+        sender_priority = self.context.sender_address.get_priority()
+        receiver_priority = self.context.local_address.get_priority()
+        return receiver_priority <= sender_priority
```

Comment:
Replace <= with <. In the unlikely event that the priorities are equal, the agent should not stop their own subproblem.